### PR TITLE
Exports for oneOf proptypes

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
Added exports on several components for proptypes using oneOf enumeration.

See issue Closes https://github.com/cerner/terra-core/issues/1044

### CODE REVIEW

File: packages/terra-badge/src/Badge.jsx [hunk 1]
```diff
@@ -6,6 +6,10 @@ import styles from './Badge.module.scss';
 
 const cx = classNames.bind(styles);
 
+const BadgeScheme = ['default', 'primary', 'secondary', 'info', 'warning', 'positive', 'negative'];
```
Comment (bjankord) [inline packages/terra-badge/src/Badge.jsx]: Let's change the name on this const to `BadgeIntent`

File: packages/terra-responsive-element/src/ResponsiveElement.jsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import ResizeObserver from 'resize-observer-polyfill';
 import 'terra-base/lib/baseStyles';
 import breakpoints from './breakpoints.module.scss';
 
+const ResponsiveElementResponsiveTo = ['window', 'parent'];
```
Comment (yuderekyu) [inline packages/terra-responsive-element/src/ResponsiveElement.jsx]: hmmmm, want to rename this? maybe `dependentViewport`

Comment (amichaelparker) [reply packages/terra-responsive-element/src/ResponsiveElement.jsx]: name changed!

**Comment by bjankord** [comment]
We'll want to keep using the original values for `PropTypes.oneOf`.

Currently, in the PR, the following type of change was made.
```diff
- intent: PropTypes.oneOf(['default', 'primary', 'secondary', 'info', 'warning', 'positive', 'negative']),
+ intent: PropTypes.oneOf([BadgeIntent.DEFAULT, BadgeIntent.PRIMARY, BadgeIntent.SECONDARY, BadgeIntent.INFO, BadgeIntent.WARNING, BadgeIntent.POSITIVE, BadgeIntent.NEGATIVE]),
```

We'll want to go with `PropTypes.oneOf` where we keep the string values or numeric values they were prior to the PR like so.
```jsx
intent: PropTypes.oneOf(['default', 'primary', 'secondary', 'info', 'warning', 'positive', 'negative']),
```

This allows people to keep using strings as values if the want.
e.g. 
```jsx
<Badge intent="primary" text="Badge text" /> or <Badge intent={BadgeIntent.PRIMARY} text="Badge text" />
```

**Pull request merged**